tests/MyEventEmitter.test.js

```diff
@@ -0,0 +1,471 @@
+const MyEventEmitter = require('../src/MyEventEmitter');
+const EventEmitter = require('events');
+
+const getRandomEventName = () => `testEvent-${Math.random()}`;
+
+const getRandomArgument = () => {
+  const types = ['string', 'number', 'boolean', 'object', 'undefined', 'null'];
+  const type = Math.floor(Math.random() * types.length);
+
+  switch (type) {
+    case 0:
+      return 'test';
+    case 1:
+      return Math.random();
+    case 2:
+      return Math.random() > 0.5;
+    case 3:
+      return {
+        test: 'test',
+        test2: Math.random(),
+        test3: Math.random() > 0.5,
+      };
+    case 4:
+      return undefined;
+    case 5:
+      return null;
+    default:
+      return undefined;
+  }
+};
+
+const getRandomArguments = () => {
+  const count = Math.floor(Math.random() * 10) + 1;
+
+  return Array.from({ length: count }, getRandomArgument);
+};
+
+describe('MyEventEmitter', () => {
+  let emitter;
+
+  beforeEach(() => {
+    emitter = new MyEventEmitter();
+  });
+
+  test('should NOT use external library dependencies', () => {
+    const sourceFile = path
+      .resolve(__dirname, '..', 'src', 'MyEventEmitter.js');
+    const fileContent = fs.readFileSync(sourceFile, 'utf8');
+    const hasCommonJSRequire = /require\(/g.test(fileContent);
+    const hasESImports = /import\s+\S+\s+from\s+['"].+['"]/g.test(fileContent);
+
+    expect(hasCommonJSRequire).toBe(false);
+    expect(hasESImports).toBe(false);
+  });
+
+  test('should NOT extend from Node.js EventEmitter', () => {
+    expect(emitter).not.toBeInstanceOf(EventEmitter);
+  });
+
+  describe('"listenerCount" method', () => {
+    test('should always return 0 for an event without listeners', () => {
+      expect(emitter.listenerCount(getRandomEventName()))
+        .toBe(0);
+    });
+  });
+
+  describe('"on" method', () => {
+    test('should add a listener for a given event and increment the listener count', () => {
+      const eventName = getRandomEventName();
+      const callback = jest.fn();
+
+      emitter.on(eventName, callback);
+
+      expect(emitter.listenerCount(eventName)).toBe(1);
+    });
+
+    test('should add multiple listeners for a given event and increment the listener count', () => {
+      const eventName = getRandomEventName();
+      const callback1 = jest.fn();
+      const callback2 = jest.fn();
+
+      emitter.on(eventName, callback1);
+      emitter.on(eventName, callback2);
+
+      expect(emitter.listenerCount(eventName)).toBe(2);
+    });
+
+    test('should add multiple listeners for multiple events and increment the listener count', () => {
+      const eventName1 = getRandomEventName();
+      const eventName2 = getRandomEventName();
+      const callback1 = jest.fn();
+      const callback2 = jest.fn();
+
+      emitter.on(eventName1, callback1);
+      emitter.on(eventName2, callback2);
+
+      expect(emitter.listenerCount(eventName1)).toBe(1);
+      expect(emitter.listenerCount(eventName2)).toBe(1);
```

Comment:
You never checked that listenerCount does not change for an event that was not handled